### PR TITLE
Tensorflow lite

### PR DESCRIPTION
**Pull request by antoinemrcr**
migrate base conversion for iOS from coreml to tensorflow-lite

### CODE REVIEW

**Comment by sunny-panchal** [comment]
Successfully converted a model and built the project on iOS with no issues; model was confirmed to work :+1: 

I'd suggest moving the instruction to install pod to the `requirements` section of the readme. Also, it was `brew install cocoapods` that I had to run

File: requirements.txt [hunk 1]
```diff
@@ -11,12 +11,16 @@ Pillow                  ==8.0.1       # HPND
...
 simpleaudio             ==1.0.4       # MIT
 
-# Dependencies needed for converting Pytorch models to CoreML
-tensorflow              ==1.14.0
-keras                   ==2.2.4
-coremltools             ==3.3
-h5py                    ==2.10.0
+# Dependencies needed for converting Pytorch models to tensorflow Lite
+tensorflow              ==2.3.1
+Keras                   ==2.4.3
```
Comment (florianletsch) [inline requirements.txt:16]: I know the file wasn't complete with licenses before, but can we agree to add license information here from now on? Both tensorflow and Keras are: `Apache 2.0`

File: requirements.txt [hunk 2]
```diff
@@ -11,12 +11,16 @@ Pillow                  ==8.0.1       # HPND
...
 pycodestyle             ==2.5.0          # MIT
-pytest                  ==5.4.3          # MIT
\ No newline at end of file
+pytest                  ==5.4.3          # MIT
+
+# Dependencies needed for converting Pytorch models to CoreML
+#tensorflow              ==1.14.0
+#keras                   ==2.2.4
+#coremltools             ==3.3
+#h5py                    ==2.10.0
```
Comment (florianletsch) [inline requirements.txt:26]: Commented out? Are these lines leftover from development and no longer needed?

File: tools/conversion/convert_to_tflite.py [hunk 3]
```diff
@@ -0,0 +1,1185 @@
...
+            outputs = []
+
+            for f in range(inputs_needed):
+                #                print('input index: ', len(all_layers) - inputs_needed + f, ' shape: ',
+                #                      all_layers[len(all_layers) - inputs_needed + f].shape)
+                inputs.append(all_layers[len(all_layers) - inputs_needed + f])
+                if merge_in > 0:
+                    #                    print('merge input index: ', len(all_layers) - inputs_needed + f, ' shape: ',
+                    #                          all_layers[len(all_layers) - (2 * inputs_needed) + f].shape)
+                    inputs.append(all_layers[len(all_layers) - (2 * inputs_needed) + f])
```
Comment (florianletsch) [inline tools/conversion/convert_to_tflite.py:245]: This script is so messy :( Will we ever clean it up?

File: requirements.txt (hunk 2 above)
Comment (antoinemrcr) [reply requirements.txt:26]: only needed if you want to a conversion to coreml. (explained in the comment)
conversion in tensorflow lite or coreml have different dependencies versions

File: tools/conversion/convert_to_tflite.py (hunk 3 above)
Comment (antoinemrcr) [reply tools/conversion/convert_to_tflite.py:245]: Not planned in this PR.

File: requirements.txt (hunk 2 above)
Comment (florianletsch) [reply requirements.txt:26]: Ah, got it. Not very easy to handle for users, though, I think (and also not testable). How about we move these dependencies to a different requirements file that can be installed in addition to the normal requirements? For example, call it `requirements-coreml.txt`?

**Review by guillaumebrg — APPROVED** [review]
Let's clean these two conversion scripts soon... I'm adding tickets to our backlog

**Pull request merged**